lib/SPIRV/SPIRVTypeScavenger.cpp

```diff
@@ -0,0 +1,561 @@
+//===- SPIRVTypeScavenger.cpp - Recover pointer types in opaque pointer IR ===//
+//
+//                     The LLVM/SPIRV Translator
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+// Copyright (c) 2022 Intel Corporation. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal with the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimers.
+// Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimers in the documentation
+// and/or other materials provided with the distribution.
+// Neither the names of Intel Corporation, nor the names of its
+// contributors may be used to endorse or promote products derived from this
+// Software without specific prior written permission.
```

Comment:
Not sure, whether that is a correct license for the translator.

Reply:
I've noticed there are a few different license block headers in this repository, and I wasn't sure which one was the proper one to use the base. If you could point to a better one, I would happily use it.

Reply:
I think that eventually we want to move to the Khronos header; you can see an example [here](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1483/files#diff-1efbc99e15b75571adb74b27faa47bbb4caf823309bb64a97888c6c8b1859a70)